modules/component-store/testing/src/testing.ts

```diff
@@ -0,0 +1,211 @@
+import { Provider, Type } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import 'jasmine';
+import { Observable, ReplaySubject } from 'rxjs';
+import {
+  MockStoreHints,
+  ProxyComponentStore,
+  SelectorKeys,
+} from './proxy-store';
+
+/**
+ * Angular providers and an injection token for a mocked component store.
+ *
+ * For usage, see {@link provideMockComponentStores}.
+ */
+export interface MockComponentStoreFactory<T> {
+  providers: Provider[];
+  token: Type<MockComponentStore<T>>;
+}
+
+interface Constructor<ClassType> {
```

Comment:
Nit/opt:  This interface allows you to infer that a constructor for a given type was passed.
- Should this be "`ConstructorOf<T>`" ?

I do not feel strongly, but I did feel like perhaps this might have sounded like a built-in, and it might be good to disambiguate.  /shruggie